### ISSUE TITLE
BeanValidator always fails when binding sub properties

### ISSUE THREAD

**Issue by Artur-** [vaadin/framework#9242]
```
public class Order {
    	@NotNull
	@OneToOne
	private Customer customer;
}
public class Customer {
	@NotNull
	@NotEmpty
	private String firstName;
}
```

```
protected TextField firstName = new TextField();

BeanValidationBinder<Order> binder = new BeanValidationBinder<>(Order.class);
binder.bind(firstName, "customer.firstName");
```

causes
```
java.lang.IllegalArgumentException: HV000039: Invalid property path. Either there is no property firstName in entity the.Order or it is not possible to cascade to the property.
```
whenever anything is added to the field.

**Comment by hesara** [comment]
Probably something that we missed in the context of #8923 - BVB.configureRequired() tries to use the correct bean, but perhaps configureBinding uses the wrong bean type for the validator.

**Comment by Artur-** [comment]
Right now it passes `Order.class` and `firstName` to `configureBinding`, which for obvious reasons won't work.

**Comment by Artur-** [comment]
Does `BeanValidator` support sub properties or should it find the sub bean type and use that?

**Comment by mlit77** [comment]
I am blocked with this issue - I can switch to vaadin 8 when this is fixed. When will it be available? Many thanks

**Comment by Artur-** [comment]
This was fixed in 8.1.0.alpha7 as indicated by the milestone
